src/v2/guide/typescript.md

```diff
@@ -0,0 +1,104 @@
+---
+title: TypeScript
+type: guide
+order: 25
+---
+
+If you develop a large application, using a language having static type checking would be helpful. If your code seems to cause unexpected behavior, it provides useful errors before you run it on browsers. You can easily use [TypeScript](https://www.typescriptlang.org/) (one of popular AltJS having static type checking) with Vue since Vue officially provides its declaration files.
+
+## Official Declaration Files
+
+There are official declaration files for Vue, vue-router and vuex. You do not have to use external tools like `typings` to use them since they are published via npm. You can install Vue's declarations with `npm install vue`. Then, all declarations can be imported with `import` statement.
+
+Example of component definition with TypeScript:
+
+``` ts
+// we cannot write `import Vue from 'vue'` in TypeScript
```

Comment:
@ryutamaki 
Actually, `import * as Vue from "vue";` and `import Vue = require("vue");` are the same thing. I choose the later syntax because it would be more intentional to import single object. 

But maybe writing the former syntax as a note would also be good. I'll add about that, thanks for pointing this out!